### PR TITLE
[NEW] Add "user is typing" view when some user is typing

### LINKED ISSUE #251: Add "user is typing" view when some user is typing

### LINKED ISSUE #690: [BUG] Allow channel unsubscribing
- Your Rocket.Chat app version: 1.4.0 (93)
- Your Rocket.Chat server version: 0.58.2
- Device (or Simulator) you're running with: Simulator iPhone 5SE

In some situations, like this, #683 , we needs unsubscribe for one channel.

Another situation is when the user change the chat channel, but still receive the messages from previous channel. For example, if you change the channel from X to Y, you still receive the messages from channel X.

### PR DESCRIPTION
**Pull request by macabeus**
@RocketChat/ios

Closes #251 
Closes #690 

- [x] Show "user is typing"
- [x] When the user typing in iOS, [must send the notify on the room](https://rocket.chat/docs/developer-guides/realtime-api/method-calls/notify-room-stream#typing)
- [x] When user change the channel, don't show "user is typing" of previous channel

### CODE REVIEW

**Comment by rafaelks** [comment]
@macabeus Thanks for your PR!. You can remove a subscription from our socket by calling this method in our web socket:

```
{"msg": "unsub", "id": "subscriptionId"}
````

**Comment by macabeus** [comment]
To continue the development of this feature, firstly needs resolve this another issue: #690
We needs a way to unsubscribe the socket when change the channel.

**Comment by macabeus** [comment]
In my last commit, I added the feature which allows the socket to unsubscribe from one channel. Then, I fixed two bugs:

- it no longer receives the messages from the previous channel (#690)
- it no longer shows "user is typing" from the previous channel

> Sorry for having broken the atomicity in commits, but I can't really split in minor parts.

**Comment by rafaelks** [comment]
@macabeus Feature is looking very good! I've just one adjustment... when the app shows the "User is typing" view, can you adjust the content offset of the collection view, so we don't block the message behind it?



Thank you!

**Comment by rafaelks** [comment]
Looking great @macabeus, thank you! Waiting for tests to merge.

**Comment by macabeus** [comment]
@rafaelks Fix!



I tried using the `collectionView?.slk_isAtBottom` to check if the user is at bottom of log, but, this function don't work, then, I created a new function to check if the user is at bottom. Maybe this new function can be used to hidden/show the `buttonScrollToBottom`?

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -261,8 +265,15 @@ final class ChatViewController: SLKTextViewController {
         scrollToBottom(true)
     }
 
-    // MARK: Message
+    override func textViewDidChange(_ textView: UITextView) {
+        if textView.text.characters.count > 0 {
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: You **could** use `isEmpty` here.

**Comment by rafaelks** [comment]
@macabeus Yes, you can use the same function. Can you do that on this PR, please?

**Comment by macabeus** [comment]
I fixed the bug in `buttonScrollToBottom` without having to use my code, then, I made another pull request: #721
And, the function that I created for this pullrequest isn't necessary in `buttonScrollToBottom`.

**Pull request merged**